ibis/expr/operations.py

```diff
@@ -3035,12 +3036,19 @@ def __repr__(self):
         )
 
     def equals(self, other, cache=None):
-        return (
+        # Check types
+        if not (
             isinstance(other, Literal)
             and isinstance(other.value, type(self.value))
-            and self.value == other.value
             and self.dtype == other.dtype
-        )
+        ):
+            return False
+
+        # Check values
```

Comment:
hmm ,does this cause a perf issue? this is actually heavy to do (though maybe just in the case of array)

Reply:
that's a good point, I think perf will not be great in this case. On the other hand, I don't think this case will be very common because this case is only hit if there are literal array scalar(s) (scalars as in "not columns") in the expression tree

Reply:
kk let's keep an eye out and if the asv's change

Reply:
Sounds good. I don't expect any perf regression from the change to this function here, because comparison between `np.arrays` didn't work at all before this change. In terms of use cases, this is meant to replace the use case where `lists` would be compared, but that has similar performance